iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergStorageHandlerMergeProperties.java

```diff
@@ -0,0 +1,86 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.mr.hive;
+
+import java.util.Properties;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat;
+import org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat;
+import org.apache.hadoop.hive.ql.io.orc.OrcInputFormat;
+import org.apache.hadoop.hive.ql.io.orc.OrcOutputFormat;
+import org.apache.hadoop.hive.ql.io.orc.OrcSerde;
+import org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat;
+import org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat;
+import org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe;
+import org.apache.hadoop.hive.ql.plan.StorageHandlerMergeProperties;
+import org.apache.hadoop.hive.serde2.avro.AvroSerDe;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.mr.Catalogs;
+
+public class IcebergStorageHandlerMergeProperties implements StorageHandlerMergeProperties {
+
+  private final Properties properties;
+
+  IcebergStorageHandlerMergeProperties(Properties properties) {
+    this.properties = properties;
+  }
+
+  public Path getTmpLocation() {
+    String location = properties.getProperty(Catalogs.LOCATION);
+    return new Path(location + "/data/");
+  }
+
+  public String getInputFileFormatClassName() {
+    FileFormat fileFormat = FileFormat.fromString(properties.getProperty(TableProperties.DEFAULT_FILE_FORMAT));
+    if (fileFormat == FileFormat.ORC) {
+      return OrcInputFormat.class.getName();
+    } else if (fileFormat == FileFormat.PARQUET) {
+      return MapredParquetInputFormat.class.getName();
+    } else if (fileFormat == FileFormat.AVRO) {
+      return AvroContainerInputFormat.class.getName();
+    }
+    return null;
+  }
+
+  public String getOutputFileFormatClassName() {
+    FileFormat fileFormat = FileFormat.fromString(properties.getProperty(TableProperties.DEFAULT_FILE_FORMAT));
+    if (fileFormat == FileFormat.ORC) {
+      return OrcOutputFormat.class.getName();
+    } else if (fileFormat == FileFormat.PARQUET) {
+      return MapredParquetOutputFormat.class.getName();
+    } else if (fileFormat == FileFormat.AVRO) {
+      return AvroContainerOutputFormat.class.getName();
+    }
+    return null;
+  }
+
+  public String getFileSerdeClassName() {
+    FileFormat fileFormat = FileFormat.fromString(properties.getProperty(TableProperties.DEFAULT_FILE_FORMAT));
+    if (fileFormat == FileFormat.ORC) {
+      return OrcSerde.class.getName();
+    } else if (fileFormat == FileFormat.PARQUET) {
+      return ParquetHiveSerDe.class.getName();
+    } else if (fileFormat == FileFormat.AVRO) {
+      return AvroSerDe.class.getName();
+    }
+    return null;
+  }
```

Comment:
nit
how about using some constant map and inner class to define this mapping and do a type check
```
  private static final class HiveFileFormat<
      I extends InputFormat<?, ?>,
      O extends OutputFormat<?,?>,
      S extends AbstractSerDe> {
    private final Class<I> inputFormatClass;
    private final Class<O> outputFormatClass;
    private final Class<S> serDeClass;

    public HiveFileFormat(Class<I> inputFormatClass, Class<O> outputFormatClass, Class<S> serDeClass) {
      this.inputFormatClass = inputFormatClass;
      this.outputFormatClass = outputFormatClass;
      this.serDeClass = serDeClass;
    }
  }

  private static final Map<FileFormat, HiveFileFormat<?,?,?>> fileFormats;

  static {
    fileFormats = new HashMap<>(3);
    fileFormats.put(FileFormat.ORC, new HiveFileFormat<>(OrcInputFormat.class, OrcOutputFormat.class, OrcSerde.class));
    fileFormats.put(FileFormat.PARQUET, new HiveFileFormat<>(MapredParquetInputFormat.class, MapredParquetOutputFormat.class, ParquetHiveSerDe.class));
    fileFormats.put(FileFormat.AVRO, new HiveFileFormat<>(OrcInputFormat.class, OrcOutputFormat.class, OrcSerde.class));
  }

  public String getInputFileFormatClassName() {
    HiveFileFormat<?, ?, ?> hiveFileFormat = fileFormats.get(FileFormat.fromString(properties.getProperty(TableProperties.DEFAULT_FILE_FORMAT)));
    return hiveFileFormat != null ? hiveFileFormat.inputFormatClass.getName() : null;
  }
```

Reply:
Replaced with existing StorageFormatDescriptor. Done.